providers/unleash/src/test/java/dev/openfeature/contrib/providers/unleash/UnleashProviderTest.java

```diff
@@ -0,0 +1,272 @@
+                .subscriber(testSubscriber)
+                .synchronousFetchOnInitialisation(synchronousFetchOnInitialisation);
+
+        UnleashOptions unleashOptions = UnleashOptions.builder()
+            .unleashConfigBuilder(unleashConfigBuilder)
+            .build();
+        return new UnleashProvider(unleashOptions);
+    }
+
+    @SneakyThrows
+    private String readBackupFile() {
+        URL url = getClass().getResource("/features.json");
+        return new String(Files.readAllBytes(Paths.get(url.toURI())));
+    }
+
+    @Test
+    void getBooleanEvaluation() {
+        assertEquals(true, unleashProvider.getBooleanEvaluation(FLAG_NAME, false, new ImmutableContext()).getValue());
+        assertEquals(true, client.getBooleanValue(FLAG_NAME, false));
+        assertEquals(false, unleashProvider.getBooleanEvaluation("non-existing", false, new ImmutableContext()).getValue());
+        assertEquals(false, client.getBooleanValue("non-existing", false));
+    }
+
+    @Test
+    void getStringVariantEvaluation() {
+        assertEquals(VARIANT_FLAG_VALUE, unleashProvider.getStringEvaluation(VARIANT_FLAG_NAME, "",
+            new ImmutableContext()).getValue());
+        assertEquals(VARIANT_FLAG_VALUE, client.getStringValue(VARIANT_FLAG_NAME, ""));
+        assertEquals("fallback_str", unleashProvider.getStringEvaluation("non-existing",
+            "fallback_str", new ImmutableContext()).getValue());
+        assertEquals("fallback_str", client.getStringValue("non-existing", "fallback_str"));
+    }
+
+    @Test
+    void getBooleanEvaluationByUser() {
+        String flagName = "by-users";
+        UnleashContext unleashContext = UnleashContext.builder().userId("111").build();
+        EvaluationContext evaluationContext = ContextTransformer.transform(unleashContext);
+        assertEquals(true, unleashProvider.getBooleanEvaluation(flagName, false, evaluationContext).getValue());
+        assertEquals(true, client.getBooleanValue(flagName, false, evaluationContext));
+        unleashContext = UnleashContext.builder().userId("2").build();
+        evaluationContext = ContextTransformer.transform(unleashContext);
+        assertEquals(false, unleashProvider.getBooleanEvaluation(flagName, false, evaluationContext).getValue());
+        assertEquals(false, client.getBooleanValue(flagName, false, evaluationContext));
+    }
+
+    @Test
+    void typeMismatch() {
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getIntegerEvaluation("test", 1, new ImmutableContext());
+        });
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getDoubleEvaluation("test", 1.0, new ImmutableContext());
+        });
+        assertThrows(TypeMismatchError.class, () -> {
+            unleashProvider.getObjectEvaluation("test", new Value(), new ImmutableContext());
+        });
+    }
+
+    @SneakyThrows
+    @Test
+    void shouldThrowIfNotInitialized() {
```

Comment:
I disagree with this test. Unleash should only throw as part of initialization if that is what the user wants.

Reply:
Please see previous [comment](https://github.com/open-feature/java-sdk-contrib/pull/424#discussion_r1321126698). End result for the user is not an exception.

Reply:
Yep this is not an exception exposed to the user. It's only used for allowing custom error handing and reporting (via hooks), and telemetry, etc.